framework/web/Response.php

```diff
@@ -758,7 +759,7 @@ public function redirect($url, $statusCode = 302, $checkAjax = true)
             $url = Yii::$app->getRequest()->getHostInfo() . $url;
         }
 
-        if ($checkAjax) {
+        if ($checkAjax && Yii::$app->getRequest()->getHeaders()->get('X-Ie-Redirect-Compatibility') !== null) {
```

Comment:
Thank you, you're right. Had a TODO to check this PR after holidays, but you came first :)
https://github.com/yiisoft/yii2/commit/019587e544f6909e7f6d2c83546fb2e063b801b9